pkg/client/dtclient/dt_client_settings.go

```diff
@@ -0,0 +1,115 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dtclient
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/log"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/client/dtclient/internal/respons"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/rest"
+	"net/http"
+	"net/url"
+)
+
+type (
+	SchemaList []struct {
+		SchemaId string `json:"schemaId"`
+	}
+
+	// SchemaListResponse is the response type returned by the ListSchemas operation
+	SchemaListResponse struct {
+		Items      SchemaList `json:"items"`
+		TotalCount int        `json:"totalCount"`
+	}
+
+	Schema struct {
```

Comment:
The client should not filter out items. We should return the Schema as-is (although currently only with a few fields) and then later filter the values we require from the schemaConstratins.
This means that the Schema is basically the 'SchemaDetail' you've defined.

Reply:
With this approach, the separation between the raw data and business structures is done here. Same as with `UpsertSettings`. There is no need to pass raw data further to the logic and make more code dependent on the outer system's raw output.

Another problem is that currently only supported type is `UNIQUE`. For others, it is not known how to unmarshaling them.

Reply:
But the client is where internal code separates from an outer system.
Passing raw data (either as a []byte or data struct) as the method's output brings nothing as a value.

On the other hand, for now, we only know the structure of the `UNIQUE` constraint, and therefore how to unmarshal it.

Reply:
But this is not 1st level client. The same is true for the automation client.
And from the technical perspective, this implementation only maps data from one structure to another. Maybe that feeling of filtration comes from the fact that only known/needed data is mapped.

Reply:
resolved in live discusion